user-service/controller/user-controller.js

```diff
@@ -101,32 +97,25 @@ export async function logoutUser(req, res) {
 
 export async function changePassword(req, res) {
     try {
-        const { username, password, newPassword } = req.body;
+        const { password, newPassword, confirmNewPassword } = req.body;
+        const username = req.user.username;
         // Check for missing fields
-        if (!username || !password || !newPassword) {
+        if (!password || !newPassword || !confirmNewPassword) {
             return res.status(400).json({message: 'Missing fields!'});
         }
-        // Check if new password and current password are the same
-        if (password === newPassword) {
-            return res.status(400).json({message: 'New password same as current password!'});
+        // Check if new password and confirm new password are the same
+        if (newPassword !== confirmNewPassword) {
+            return res.status(400).json({message: 'The passwords you entered do not match!'});
         }

```

Comment:
I'm good with this but it seems that it is recommended to handle both of these in frontend.

https://stackoverflow.com/questions/42190530/matching-passwords-with-front-end-or-back-end

Just for awareness.

Reply:
Solved

Confirm password will be checked in frontend for both sign up and change password functions.

![image](https://user-images.githubusercontent.com/77485021/193453514-c132dd38-7458-4228-b9aa-423d6b5c075b.png)